### DISCUSSION TITLE
App Layout: Consider pinning favourite rooms to top

### DISCUSSION THREAD

**Discussion by noantiq** [element-hq/element-meta#846] [Product Feedback]
### Your use case

First of all I want to say, that I am really happy to see the work happening for the new app layout.

#### What would you like to do?
Easily navigate to my favourite room(s) in a familiar manner.
#### Why would you like to do it?
Rooms I marked as favourites are special to me and should be found quickly without scrolling the room list or searching for them.
#### How would you like to achieve it?
I would like to have my favourite rooms pinned to the top of my rooms list. It feels familiar since it is the same way of letting special rooms stand out like in Element Web, the default layout of Element Android, FluffyChat, Signal, WhatsApp, Telegram and probably many more.
I basically would like to have have my favourites displayed like in the default layout, see:
![Screenshot_20220911-125051](https://user-images.githubusercontent.com/39308834/189523700-4be06953-8d86-4906-b99e-7f89ee87f650.png)
, but have my rooms and DMs merged like in the new layout.

### Have you considered any alternatives?

The new app layout gives the option to display filters. Like this it is possible to access Favourites using the tab. For me that option is rather inconvenient and just a waste of space for accessing my (single) favourite room. See:
![Screenshot_20220911-124343](https://user-images.githubusercontent.com/39308834/189523421-e2007725-e522-41e4-aea7-638be15dcd30.png)

### Additional context

I experienced a very similar problem with the rooms marked as low priority. They are not pinned to the bottom of my rooms list as expected but mixed into the room list like my favourite room(s). See https://github.com/vector-im/element-android/issues/7089

### Are you willing to provide a PR?

No

**Comment by kojid0** [comment]
Instead of having another tab for favorites, they can simply be pinned to top, no matter whether _recents_ or _filters_ are enabled or not.
This will save space by having one less tab and can solve issues like vector-im/element-android#7155.

![grafik](https://user-images.githubusercontent.com/24477557/191240546-8bf9116b-d971-4ea4-9b87-98172f735bff.png)

**Comment by kojid0** [comment]
Another option would be to show favorites in the horizontal bar. If the user decides to enable the horizontal top bar, he can decide between favorites or recents.

- 1) User doesn't enable top bar: 
List favorites as pinned to top (like [here](https://github.com/vector-im/element-meta/issues/845))
- 2) User enables top bar and chooses recents: 
List favorites as pinned to top (like [here](https://github.com/vector-im/element-meta/issues/845))
- 3) User enables top bar and chooses favorites:
List favorites in top bar like this: 
![grafik](https://user-images.githubusercontent.com/24477557/191245878-edbdc15b-5687-406b-9cb9-84e3c2b8abc1.png)

**Comment by kojid0** [comment]
Pinging the design team (@daniellekirkwood ?) as this should be labeled as a blocker before the new app layout gets released.

**Comment by daniellekirkwood** [comment]
Thanks for the feedback! 

This isn't a blocker to our first release but we really appreciate the feedback and will be looking to address Favourites and Low Priority in the future.

**Comment by NoraCodes** [comment]
Hi! I just received this update and wanted to mention that having no support for favorites or low priority rooms is an enormous issue for me. As someone with ADHD it makes using Element on mobile very difficult for me. I would appreciate it if removing important features was considered a bigger deal in the future!

For context, I run a homeserver for my friends and family, and Matrix via Element is my main method of communication with them, so it's a pretty big issue in my life.

**Comment by codesalatdev** [comment]
The new layouts looks fantastic and solves the issue of having to switch tabs between group chats and DMs. But straight up ignoring favorites and low priority creates issues that we didn't have in the first place.

Both would be perfect, neatly packaged into one layout/screen as suggested here before. Users are accustomed to this layout by other messenger apps and believe it or not, this has become a reason why people I recommend element to decided against it (The schildi fork solves the first issue, but the new layout looks just so much better).

**Comment by daniellekirkwood** [comment]
I'm going to be moving this to Discussions on Element meta as the feedback provided should be taken into consideration on both iOS and Android as we continue to improve on our first launch. 

Thanks so much!

**Comment by franzitrone** [comment]
Is this still in consideration?

**Comment by lfstuttgart** [comment]
I would really like to see that happen

**Comment by awtj8o81ryywg793** [comment]
Now that we’ve (mostly) transitioned from Element Classic to Element X, and considering the refinement of the design illustrated in the second screenshot with a pill-based filter layout and mechanism, I’m unsure how compatible this idea is with the new layout.

You could remove the favourites filter and place favourite rooms at the top of the (un-)filtered room list; however, this might lead to navigation issues if a user has a considerable number of favourite rooms.

**Reply by ian-hls2** [reply]
I don't like the new pill-based filter system with regard to favourites because it seems like you favourite chats so they are easy to get to. Having to click the arrow to see more filters and then click favourites is an extra step. I like the idea of just having favourites pinned to the top of the chat (that should at least be an option in preferences). Another option might be to allow the filter pills to be rearranged by the user so they can put favourites first or second.

**Reply by lfstuttgart** [reply]
In my opinion in every application featuring them favourites are elements accessed very very frequently. Therefore they should be only one click away. Any design should cater for that.
Maybe the Favourites pill can be the first or second pill? That way on most window sizes it would not hide behind the arrow.

**Reply by awtj8o81ryywg793** [reply]
Quoted from a [previous reply](https://github.com/element-hq/element-meta/discussions/846#discussioncomment-14944498):

> Maybe the Favourites pill can be the first or second pill? That way on most window sizes it would not hide behind the arrow.

I'm not against such an idea.

**Reply by MarianRychtecky** [reply]
> I don't like the new pill-based filter system with regard to favourites because it seems like you favourite chats so they are easy to get to. Having to click the arrow to see more filters and then click favourites is an extra step. I like the idea of just having favourites pinned to the top of the chat (that should at least be an option in preferences). Another option might be to allow the filter pills to be rearranged by the user so they can put favourites first or second.

I agree, the new pill-based filter is an unnecessary burden when it comes to quick navigation. A pinned room with a fixed position (order) will do the job perfectly. I would like to see the addition of an Archive feature (simply marking old conversations as archived).
I agree, at least allowing rearranging the pills would help, but not solve.

**Comment by MarianRychtecky** [comment]
I'm a big supporter of this feature. And I would like to add - please ensure that pinned rooms do not change their order with messages received.
